modules/apps/commerce/commerce-product-type-virtual-web/src/main/resources/META-INF/resources/js/TermsOfUseInputUtil.js

```diff
@@ -0,0 +1,152 @@
+					}
+
+					journalArticleRemove.classList.remove('hide');
+
+					journalArticleNameInput.innerText = itemValue.title;
+					if (Liferay.FeatureFlags['LPD-11235']) {
+						const ckEditorToolbar = document.querySelector(
+							'#termsOfUseContent .ck-editor__top'
+						);
+						ckEditorToolbar.style.display = 'none';
+
+						const ckEditorContent = document.querySelector(
+							'#termsOfUseContent .ck-editor__main .ck-content'
+						);
+						ckEditorContent.classList.add('ck-read-only');
+						ckEditorContent.setAttribute('contenteditable', false);
+					}
+				},
+				selectEventName: 'selectJournalArticle',
+				title: Liferay.Language.get('select-web-content'),
+				url: termsOfUseJournalArticleBrowserURL,
+			});
+		});
+
+		journalArticleRemove.addEventListener('click', (event) => {
+			event.preventDefault();
+
+			const termsOfUseJournalArticleResourcePrimKey =
+				document.getElementById(
+					`${portletNamespace}termsOfUseJournalArticleResourcePrimKey`
+				);
+
+			if (termsOfUseJournalArticleResourcePrimKey) {
+				termsOfUseJournalArticleResourcePrimKey.value = 0;
+			}
+
+			journalArticleNameInput.innerText = Liferay.Language.get('none');
+
+			journalArticleRemove.classList.add('hide');
+
+			if (Liferay.FeatureFlags['LPD-11235']) {
+				const ckEditorToolbar = document.querySelector(
+					'#termsOfUseContent .ck-editor__top'
+				);
+				ckEditorToolbar.style.display = 'block';
+
+				const ckEditorContent = document.querySelector(
+					'#termsOfUseContent .ck-editor__main .ck-content'
+				);
+				ckEditorContent.classList.remove('ck-read-only');
+				ckEditorContent.setAttribute('contenteditable', true);
+			}
+		});
+	}
+	if (Liferay.FeatureFlags['LPD-11235']) {
+		window.onload = () => {
+			waitForElement('#termsOfUseContent .ck-editor__top').then(() => {
+				evaluateContentVisibility();
+			});
+		};
+
+		const evaluateContentVisibility = () => {
+			const ckEditorToolbar = document.querySelector(
+				'#termsOfUseContent .ck-editor__top'
+			);
+
+			const hasTermsOfUseContent = useTermsOfUseJournal === true;
+
+			if (hasTermsOfUseContent) {
+				ckEditorToolbar.style.display = 'none';
+				const ckEditorContent = document.querySelector(
+					'#termsOfUseContent .ck-editor__main .ck-content'
+				);
+				ckEditorContent.classList.add('ck-read-only');
+				ckEditorContent.setAttribute('contenteditable', false);
+			}
+			else {
+				ckEditorToolbar.style.display = 'block';
+				const ckEditorContent = document.querySelector(
+					'#termsOfUseContent .ck-editor__main .ck-content'
+				);
+				ckEditorContent.classList.remove('ck-read-only');
+				ckEditorContent.setAttribute('contenteditable', true);
+			}
+		};
+
+		const waitForElement = (selector) => {
+			return new Promise((resolve) => {
+				const selectedElement = document.querySelector(selector);
+				if (selectedElement) {
+					return resolve(selectedElement);
+				}
+
+				const observer = new MutationObserver(() => {
```

Comment:
Why do we need such an intensive handler? Is this demanded by the new InputLocalized tag usage?